common/codegen/src/writes/arithmetics_type_v2.rs

```diff
@@ -55,16 +55,36 @@ pub fn codegen_arithmetic_type_v2() {
 // This code is generated by common/codegen. DO NOT EDIT.
 use super::number::Number;
 
-pub trait ResultTypeOfBinary {{
+pub trait ResultTypeOfBinary: Sized {{
     type AddMul: Number;
     type Minus: Number;
     type IntDiv: Number;
     type Modulo: Number;
     type LeastSuper: Number;
 }}
 
-pub trait ResultTypeOfUnary {{
+pub trait ResultTypeOfUnary: Sized {{
     type Negate: Number;
+    
+    fn d_checked_add(self, _rhs: Self) -> Option<Self> {{
```

Comment:
what does the `d_` mean?

Reply:
It's just to be different with the built-in function `checked_add` in std lib.

Reply:
It's weird. Can we simply name it `checked_add` just like num-traits https://docs.rs/num-traits/0.2.15/num_traits/ops/checked/trait.CheckedAdd.html